routing/src/nexthop.rs

```diff
@@ -14,19 +13,19 @@ use std::net::IpAddr;
 use std::option::Option;
 #[cfg(test)]
 use std::str::FromStr;
-use std::sync::Arc;
+use std::sync::{Arc, RwLock};
```

Comment:
Just flagging the existing discussion with Manish about the locks

Reply:
Yes. I'm not too concerned about deadlocks ... yet, because the writers are verly localized and we mostly read elsewhere.

Reply:
I think this lock usage is fine for these reasons.  And because they are contained and don't lead out of this stage, I think it is fine.  But we need to watch this, it could quickly spiral out of control.  I want @daniel-noland to look at this and propose alternatives though.  If there is an easy way to avoid this locking, that would be preferable.